src/index.ts

```diff
@@ -47,9 +47,19 @@ async function disableSparseCheckout() {
     console.log('Removed NXGIT controlled checkout. Now you can use git!!');
 }
 
-async function addApp(name: string) {
-    executeCommand(`git sparse-checkout add ${name}`);
-    console.log(`Added app: ${name}`);
+async function addApp(name: string, options: any) {
+    if (options.folderOnly) {
+        executeCommand(`git sparse-checkout add ${name}`);
+        console.log(`Added folder: ${name}`);
+    } else {
+        const dependencies = getAppDependencies(name);
+        if (dependencies) {
+            dependencies.forEach(dep => {
+                executeCommand(`git sparse-checkout add ${dep.path}`);
```

Comment:
Can you please add error handling and notify in the console, if the folder cannot be added or the folder name is not right?